ts/Extensions/Annotations/Popup.ts

```diff
@@ -932,6 +977,140 @@ H.Popup.prototype = {
                 (indicatorList.childNodes[0] as any).click();
             }
         },
+        /**
+         * Add selection HTML element and its' label.
+         *
+         * @private
+         *
+         * @param {string} indicatorType
+         *        Type of the indicator i.e. sma, ema...
+         *
+         * @param {string} [optionName]
+         *        Name of the option into which selection is being added.
+         *
+         * @param {HTMLDOMElement} [parentDiv]
+         *        HTML parent element.
+         *
+         * @return {HTMLSelectElement}
+         */
+        addSelection: function (
+            this: Highcharts.Popup,
+            indicatorType: string,
+            optionName: string,
+            parentDiv: HTMLDOMElement
+        ): HTMLSelectElement {
+            const optionParamList = optionName.split('.'),
+                labelText = optionParamList[optionParamList.length - 1];
+            let selectName = PREFIX + optionName + '-type-' + indicatorType,
+                lang = this.lang,
+                selectBox: HTMLSelectElement;
+
+            // Add a label for the selection box.
+            createElement(
+                LABEL, {
+                    htmlFor: selectName
+                },
+                null as any,
+                parentDiv
+            ).appendChild(doc.createTextNode(
+                lang[labelText] || optionName
+            ));
+
+            // Create a selection box.
+            selectBox = createElement(
+                SELECT,
+                {
+                    name: selectName,
+                    className: PREFIX + 'popup-field'
+                },
+                null as any,
+                parentDiv
+            ) as HTMLSelectElement;
+
+            selectBox.setAttribute('id', PREFIX + 'select-' + optionName);
```

Comment:
Can we add the `id` inside `createElement` instead of calling `setAttribute` again?